plugins/link/SharedMemory.cpp

```diff
@@ -146,11 +150,47 @@ void *SharedMemory::mapMemory(const char *name, std::size_t size) {
 	m_name.assign(name);
 #endif
 
-	m_size = size;
-
 	if (created) {
-		std::memset(m_data, 0, m_size);
+		fillWithZero();
+	}
+
+	return true;
+}
+
+bool SharedMemory::isMemoryMapped() {
+	return m_data != nullptr;
+}
+
+void SharedMemory::read(LinkedMem *dest) {
+	std::atomic< Chunk > *data = (std::atomic< Chunk > *) m_data;
+
+	Chunk buff[chunkCount];
+
+	for (std::size_t i = 0; i < chunkCount; i++) {
+		buff[i] = data[i].load(std::memory_order_relaxed);
+	}
+
+	// bitcast the chunks with memcpy to avoid running afoul
+	// of type-based alias analysis
+	std::memcpy(dest, buff, sizeof(LinkedMem));
+}
+
+void SharedMemory::write(const LinkedMem *source) {
+	std::atomic< Chunk > *data = (std::atomic< Chunk > *) m_data;
```

Comment:
```suggestion
	std::atomic< Chunk > *data = reinterpret_cast< std::atomic< Chunk > *>(m_data);
```
(same comment applies as to the `read?  implementation)